File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/entities/Component.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "component")
public class Component {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comp_id")
    private int compId;

    @Column(name = "comp_name", nullable = false)
    private String compName;

    public Component() {
    }

    public Component(int compId, String compName) {
        this.compId = compId;
        this.compName = compName;
    }

    public int getCompId() {
        return compId;
    }

    public void setCompId(int compId) {
        this.compId = compId;
    }

    public String getCompName() {
        return compName;
    }

    public void setCompName(String compName) {
        this.compName = compName;
    }

}

File: backend-dotnet/Models/Component.cs
//using System.ComponentModel.DataAnnotations;
//using System.ComponentModel.DataAnnotations.Schema;

//namespace backend_dotnet.Models
//{
//    [Table("component")]
//    public class Component
//    {
//        [Key]
//        [Column("comp_id")]
//        public int CompId { get; set; }

//        [Required]
//        [Column("comp_name")]
//        public string CompName { get; set; } = string.Empty;
//    }
//}



using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models
{
    [Table("component")]
    public class Component
    {
        [Key]
        [Column("comp_id")]
        public int CompId { get; set; }

        [Required]
        [Column("comp_name")]
        public string CompName { get; set; } = string.Empty;
    }
}
